### DISCUSSION TITLE
Event Listener not working

### DISCUSSION THREAD

**Discussion by rmitesh** [livewire/livewire#6730] [1. Help]
In Livewire 3 I have 2 components `ScreenA` and `ScreenB`.

I am sending `ScreenA`'s `Post` data to `ScreenB`

```php
class ScreenA extends Component
{
    public function send(): void
    {
        $post = Post::first();
        $this->dispatch('post-created', title: $post->title)->to(ScreenB::class);
    }
}
```

ScreenA View file

```html
<button wire:click="send">Send</button>
```

Screen B
```php
class ScreenB extends Component
{
    public function render()
    {
        return view('livewire.screen-b');
    }
    
    #[On('post-created')]
    public function listenAEvents($post): void
    {
        $this->data[] = $post;
    }
}

```
and in the screen-b view file I am displaying that data

```php
<ol>
    @foreach( $data as $value )
        <li>{{ $value }}</li>
    @endforeach
</ol>
```

anyone help me what wrong in the code?

**Comment by danie-ramdhani** [comment]
did you forget to add this?
```php
use Livewire\Attributes\On;
```

**Reply by rmitesh** [reply]
Nope, it's already imported

**Comment by misenhei** [comment]
Is your listening function not catching the dispatch at all, or is it catching it but $post is not set? If $post not being set is the issue try changing 
$this->dispatch('post-created', title: $post->title)->to(ScreenB::class);
To
$this->dispatch('post-created', post: $post->title)->to(ScreenB::class);
I am pretty sure the parameter name needs to match the variable name in your listener.

I may be having a similar issue. I upgraded from v2 to v3 and used automated upgrade along with the steps to replace emit with dispatch. My listeners are catching the dispatches, but data is not being received by listeners. I will wait to see how this gets resolved before I either start a new discussion or tag more on to this one.

**Reply by rmitesh** [reply]
Nope, I have tried to change the variable name to the listener parameter, but still not working

**Reply by misenhei** [reply]
So is you listening function listenAEvents catching the dispatch and your variable is just not set, or is it not being triggered at all?

**Comment by misenhei** [comment]
I have the same issue. After more testing my listeners are not catching the dispatches. I thought they were earlier and I was just missing data but I am mistaking. They are not catching at all. I upgraded from Livewire v2, and prior to doing so my emits worked without issue. Post upgrading to v3:

In sender class I replaced:
`$this->emit('updateEmailBlockRule', $r_id);`
with:
`$this->dispatch('updateEmailBlockRule', rule_id: $r_id);`

In receiver class I
Added: 
`use Livewire\Attributes\On;`
Deleted: 
```
protected $listeners = [
        'updateEmailBlockRule' => 'mount',
    ];
```
Added:
```
    #[On('updateEmailBlockRule')] 
    public function mount( $rule_id = null )
    {....
```

Anyone else having this problem or know what we are doing wrong? Please help.

**Reply by posb00** [reply]
i got the same issue, there is a solution for this?

**Reply by PhiloNL** [reply]
Did you import the entire namespace? `use Livewire\Attributes\On;`

**Reply by posb00** [reply]
yes of curse! but still not working
